### PR TITLE
HW#2: Webpack module

### PR DESCRIPTION
**Pull request by Marysosh**
### Homework summary

In scope of this homework I've created a new version of the application using webpack and set up linters to check code style.

Homework in details:

1.  Now there are 2 folders with applications: NetflixRouletteCreatereactApp(old version) and NetflixRouletteWebpack(new version)

2. Webpack configuration details:

- There are 3 main webpack config files: webpack.common.js, webpack.dev.js and webpack.prod.js: 
       - webpack.common.js uses [babel loader](https://webpack.js.org/loaders/babel-loader/)
       - webpack.dev.js uses CSS loaders([sass-loader](https://webpack.js.org/loaders/sass-loader/), [css-loader](https://webpack.js.org/loaders/css-loader/), [style-loader](https://webpack.js.org/loaders/style-loader/)) and [html-webpack-plugin](https://webpack.js.org/plugins/html-webpack-plugin/)
       - webpack.prod.js uses minimizers([css-minimizer-plugin](https://webpack.js.org/plugins/css-minimizer-webpack-plugin/),[terser-plugin](https://webpack.js.org/plugins/terser-webpack-plugin/)), plugins([clean-webpack-plugin](https://github.com/johnagan/clean-webpack-plugin),[mini-css-extract](https://github.com/webpack-contrib/mini-css-extract-plugin),[html-webpack-plugin](https://webpack.js.org/plugins/html-webpack-plugin/)) and CSS loaders([sass-loader](https://webpack.js.org/loaders/sass-loader/), [css-loader](https://webpack.js.org/loaders/css-loader/), [mini-css-extract-plugin.loader](https://github.com/webpack-contrib/mini-css-extract-plugin))
- There is a script for running application with dev server(**npm start**) and one another script for creating production build(**npm run build**). The Webpack config file for this script is picked up automatically and depends on the env variable.

3. Eslint and prettier code linters are also configured. There is a husky pre-commit hook which works with lint-staged and catches codestyle errors.

### CODE REVIEW

File: NetflixRouletteWebpack/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (NikitaUstsiushenka) [inline NetflixRouletteWebpack/.gitignore:1]: @Marysosh Please add new line in end of file

File: NetflixRouletteWebpack/src/components/counter/Counter.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class Counter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: 0 };
+  }
+
+  changeValue(increase) {
+    const { value } = this.state;
+    this.setState(increase ? { value: value + 1 } : { value: value - 1 });
```
Comment (NikitaUstsiushenka) [inline NetflixRouletteWebpack/src/components/counter/Counter.js:12]: @Marysosh You can simplify like `value++` and `value--`

Comment (mariia-soshnikova-EPAM) [reply NetflixRouletteWebpack/src/components/counter/Counter.js:12]: I tried and it breaks the functionality, the counter stops working. Moreover, eslint throws an exception:
image

File: NetflixRouletteWebpack/.gitignore (hunk 1 above)
Comment (mariia-soshnikova-EPAM) [reply NetflixRouletteWebpack/.gitignore:1]: Corrected

**Pull request merged**